org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java

```diff
@@ -402,6 +403,26 @@ public boolean isCompletionListItemDefaultsInsertTextFormatSupport() {
 			&& capabilities.getTextDocument().getCompletion().getCompletionList().getItemDefaults().contains("insertTextFormat");
 	}
 
+	public boolean isCompletionListItemDefaultsInsertTextModeSupport() {
+		return v3supported
+			&& capabilities.getTextDocument().getCompletion() != null
+			&& capabilities.getTextDocument().getCompletion().getCompletionList() != null
+			&& capabilities.getTextDocument().getCompletion().getCompletionList().getItemDefaults() != null
+			&& capabilities.getTextDocument().getCompletion().getCompletionList().getItemDefaults().contains("insertTextMode");
+	}
+
+	public boolean isCompletionItemInsertTextModeAdjustIndentationSupport() {
```

Comment:
How about changing sth like `boolean isCompletionItemInsertTextModeSupport(String insertMode)`. Then we are using one function dealing **N** different types of insert mode check.

Same change can be done to the above `isCompletionListItemDefaultsInsertTextModeSupport()`